### PR TITLE
Warn users if Docker arch does not match host arch

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
When running SQL Server on M1 Mac:

Screenshot 2023-02-21 at 16 34 07

When running MinIO (it has dedicated ARM image):

Screenshot 2023-02-21 at 16 34 48

File: testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                    Detected incompatible Docker image and host architectures. The performance of running docker images in such scenarios can vary or not work at all.
+                    Host: %s (%s).
+                    Docker architecture: %s.
+                    Docker image architecture: %s.
+
+                    Set environment variable TESTCONTAINERS_SKIP_ARCHITECTURE_CHECK=true to skip this check.
+                    !!! WARNING !!!
+                    """.formatted(System.getProperty("os.name"), System.getProperty("os.arch"), dockerArch, imageArch));
+
+            exit(99);
```
Comment (nineinchnick) [inline testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java:83]: It's not a warning if it exits, but an error, right?

Comment (wendigo) [reply testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java:83]: Correct

Comment (nineinchnick) [reply testing/trino-testing-containers/src/main/java/io/trino/testing/containers/PlatformChecks.java:83]: Can you then change `!!! WARNING !!!` to `!!! ERROR !!!`?

**Review by nineinchnick — APPROVED** [review]

**Review by vlad-lyutenko — APPROVED** [review]

**Pull request merged**